test/helpers/votingTestCases.js

```diff
@@ -0,0 +1,188 @@
+import { expect } from "chai";
+import { increaseTime } from "../helpers/evmCommands";
+import { createSignedVote } from "../helpers/relayedVoteSigning";
+import { prettyPrintGasCost } from "../helpers/gasUtils";
+
+export function testVotingWithSnapshots(getToken, getVotingContract, owner, owner2, relayer) {
+  let token;
+  let votingContract;
+
+  beforeEach(() => {
+    votingContract = getVotingContract();
+    token = getToken();
+  });
+
+  describe("Opening Proposals:", () => {
+    it("should allow to open a proposal", async () => {
```

Comment:
practical note:
you should also test public properties of the contract after it has been deployed. is `TOKEN` set correctly? is voting period as expected?